dvc/fs/memory.py

```diff
@@ -1,34 +1,35 @@
-from .base import BaseFileSystem
+import threading
 
+from funcy import cached_property, wrap_prop
 
-class MemoryFileSystem(BaseFileSystem):
-    scheme = "local"
-    PARAM_CHECKSUM = "md5"
-
-    def __init__(self, **kwargs):
-        from fsspec.implementations.memory import MemoryFileSystem as MemFS
+from dvc.path_info import CloudURLInfo
+from dvc.scheme import Schemes
 
-        super().__init__(**kwargs)
+from .fsspec_wrapper import FSSpecWrapper
 
-        self.fs = MemFS()
 
-    def exists(self, path_info) -> bool:
-        return self.fs.exists(path_info.fspath)
-
-    def open(self, path_info, mode="r", encoding=None, **kwargs):
-        return self.fs.open(
-            path_info.fspath, mode=mode, encoding=encoding, **kwargs
-        )
+class MemoryFileSystem(FSSpecWrapper):  # pylint:disable=abstract-method
+    scheme = Schemes.MEMORY
+    PARAM_CHECKSUM = "md5"
+    PATH_CLS = CloudURLInfo
+    TRAVERSE_PREFIX_LEN = 2
+    DEFAULT_BLOCKSIZE = 4096
```

Comment:
for whatever reason, fsspec's MemFile file-object does not define its own default blocksize, but our fsspec wrapper expects all of the cloud fobj's to define them